BeagleBone_black/src/state_machine/main.cpp

```diff
@@ -83,12 +83,38 @@ void Main::checkCommunications()
     hypedMachine.reset();
   }
 }
+
+void Main::checkFailure()
+{
+  if (comms_data.module_status == data::ModuleStatus::kCriticalFailure
+      || nav_data.module_status == data::ModuleStatus::kCriticalFailure
+      /*|| motor_data.module_status == data::ModuleStatus::kCriticalFailure*/
+      || sensors_data.module_status == data::ModuleStatus::kCriticalFailure
+      || batteries_data.module_status == data::ModuleStatus::kCriticalFailure) {
+    hypedMachine.handleEvent(kCriticalFailure);  //  Transitions to FailureStopped/EmergencyBraking
+  }
+}
+
 //  @TODO add checks for other modules' states
 void Main::checkReady()
 {
-  if (nav_data.state == data::NavigationState::kReady) {
-    hypedMachine.handleEvent(kSystemsChecked);
+  if (nav_data.module_status == data::ModuleStatus::kReady
+  /*&&  motor_data.module_status == data::ModuleStatus::kReady*/) {
+    hypedMachine.handleEvent(kSystemsChecked);  // Transitions to Ready
   }
 }
 
+void Main::checkInit()
+{
+  if (comms_data.module_status == data::ModuleStatus::kInit
+      || nav_data.module_status == data::ModuleStatus::kInit
+      /*|| motor_data.module_status == data::ModuleStatus::kInit*/
+      || sensors_data.module_status == data::ModuleStatus::kInit
+      || batteries_data.module_status == data::ModuleStatus::kInit) {
```

Comment:
I believe this should be && not ||, i.e. all modules must be in kInit for the state machine to transition to calibrating

Reply:
Oh ups, that's true .

Reply:
also, why is the check for motor_data commented out?